src/main/java/org/opensearch/flowframework/workflow/RegisterModelStep.java

```diff
@@ -0,0 +1,154 @@
+import org.opensearch.ml.common.transport.register.MLRegisterModelInput;
+import org.opensearch.ml.common.transport.register.MLRegisterModelResponse;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+/**
+ * Step to register a remote model
+ */
+public class RegisterModelStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(RegisterModelStep.class);
+
+    private NodeClient nodeClient;
+
+    static final String NAME = "register_model";
+
+    private static final String FUNCTION_NAME = "function_name";
+    private static final String MODEL_NAME = "name";
+    private static final String MODEL_VERSION = "model_version";
+    private static final String MODEL_GROUP_ID = "model_group_id";
+    private static final String DESCRIPTION = "description";
+    private static final String CONNECTOR_ID = "connector_id";
+    private static final String MODEL_FORMAT = "model_format";
+    private static final String MODEL_CONFIG = "model_config";
+
+    /**
+     * Instantiate this class
+     * @param nodeClient client to instantiate MLClient
+     */
+    public RegisterModelStep(NodeClient nodeClient) {
+        this.nodeClient = nodeClient;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+
+        CompletableFuture<WorkflowData> registerModelFuture = new CompletableFuture<>();
+
+        MachineLearningNodeClient machineLearningNodeClient = MLClient.createMLClient(nodeClient);
+
+        ActionListener<MLRegisterModelResponse> actionListener = new ActionListener<>() {
+            @Override
+            public void onResponse(MLRegisterModelResponse mlRegisterModelResponse) {
+                logger.info("Model registration successful");
+                registerModelFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(
+                            Map.entry("modelId", mlRegisterModelResponse.getModelId()),
+                            Map.entry("model-register-status", mlRegisterModelResponse.getStatus())
+                        )
+                    )
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to register model");
+                registerModelFuture.completeExceptionally(new IOException("Failed to register model "));
+            }
+        };
+
+        FunctionName functionName = null;
+        String modelName = null;
+        String modelVersion = null;
+        String modelGroupId = null;
+        String connectorId = null;
+        String description = null;
+        MLModelFormat modelFormat = null;
+        MLModelConfig modelConfig = null;
+
+        for (WorkflowData workflowData : data) {
+            if (workflowData != null) {
```

Comment:
again, answered above :)

Reply:
Re-commenting here.   If you're getting `null` workflow data from anywhere, that's a bug.  `WorkflowData.EMPTY` exists for precisely the case where you have nothing to send.

We should not need null checks.  If we are encountering null here, change whatever is sending it to send `WorkflowData.EMPTY` instead of null.